pystac/extensions/pointcloud.py

```diff
@@ -0,0 +1,347 @@
+            self.item.properties['pc:encoding'] = encoding
+        else:
+            asset.properties['pc:encoding'] = encoding
+
+    @property
+    def schemas(self):
+        """Get or sets a
+
+        The schemas represent the structure of the data attributes in the pointcloud,
+        and is represented as a sequential array of items that define the dimensions
+        and their types,
+
+        Returns:
+            List[dict]
+        """
+        return self.get_schemas()
+
+    @schemas.setter
+    def schemas(self, v):
+        self.set_schemas(v)
+
+    def get_schemas(self, asset=None):
+        """Gets an Item or an Asset projection geometry.
+
+        If an Asset is supplied and the Item property exists on the Asset,
+        returns the Asset's value. Otherwise returns the Item's value
+
+        Returns:
+            dict
+        """
+        if asset is None or 'pc:schema' not in asset.properties:
+            return self.item.properties.get('pc:schemas')
+        else:
+            return asset.properties.get('pc:schemas')
+
+    def set_schemas(self, schemas, asset=None):
+        """Set an Item or an Asset schema
+
+        If an Asset is supplied, sets the property on the Asset.
+        Otherwise sets the Item's value.
+        """
+        if asset is None:
+            self.item.properties['pc:schemas'] = schemas
+        else:
+            asset.properties['pc:schemas'] = schemas
+
+    @property
+    def density(self):
+        """Get or sets the density for the item.
+
+        Density is defined as the number of points per square unit area.
+
+        Returns:
+            int
+        """
+        return self.get_density()
+
+    @density.setter
+    def density(self, v):
+        self.set_density(v)
+
+    def get_density(self, asset=None):
+        """Gets an Item or an Asset density.
+
+        If an Asset is supplied and the Item property exists on the Asset,
+        returns the Asset's value. Otherwise returns the Item's value
+
+        Returns:
+            int
+        """
+        if asset is None or 'pc:density' not in asset.properties:
+            return self.item.properties.get('pc:density')
+        else:
+            return asset.properties.get('pc:density')
+
+    def set_density(self, density, asset=None):
+        """Set an Item or an Asset density property.
+
+        If an Asset is supplied, sets the property on the Asset.
+        Otherwise sets the Item's value.
+        """
+        if asset is None:
+            self.item.properties['pc:density'] = density
+        else:
+            asset.properties['pc:density'] = density
+
+    @property
+    def statistics(self):
+        """Get or sets the statistics for each property of the dataset.
+
+        A sequential array of items mapping to pc:schemas defines per-channel statistics.
+
+        Exmample::
+
+            item.ext.pointcloud.statistics = [{ 'name': 'red', 'min': 0, 'max': 255 }]
+
+        Returns:
+            List[dict]
```

Comment:
Same comment as with the `schemas` property - consider using a specific object for so that users have clear guidance on how to construct or use what's returned here.

Reply:
add PointcloudStatistic to handle this.